com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -1081,4 +1083,28 @@ public static void setServicesGuids(DelegateExecution context, Map<String, Strin
         }.getType());
     }
 
+    public static CloudServiceExtended getServiceToProcess(DelegateExecution context) {
+        String service = (String) context.getVariable(Constants.VAR_SERVICE_TO_PROCESS);
+        if (StringUtils.isEmpty(service)) {
+            return null;
+        }
+        return (CloudServiceExtended) JsonUtil.fromJson(service, CloudServiceExtended.class);
+    }
+
+    public static void setServiceActions(List<ServiceAction> actions, DelegateExecution context) {
+        List<String> actionsNames = actions.stream()

```

Comment:
Why doing this for the enums? Would it be sufficient to just persist the list in the context?